src/server/main_service.cc

```diff
@@ -2029,6 +2030,11 @@ GlobalState Service::GetGlobalState() const {
 }
 
 void Service::ConfigureHttpHandlers(util::HttpListenerBase* base) {
+  // We set the password for the HTTP service unless it is only enabled on the
+  // admin port and the admin port is password-less.
+  if (GetFlag(FLAGS_primary_port_http_enabled) || !GetFlag(FLAGS_admin_nopass)) {
```

Comment:
Why this logic?
Why not use `--requirepass` for non-admin HTTP connections?
Is the admin-port HTTP page different than the non-admin one?

Reply:
`GetPassword()` checks `requirepass` internally so it's used for both admin and non-admin connections. They're the same HTTP page.

Reply:
Gotcha